Core/src/org/sleuthkit/autopsy/coreutils/VideoUtils.java

```diff
@@ -148,48 +160,77 @@ static BufferedImage generateVideoThumbnail(AbstractFile file, int iconSize) {
+                    LOGGER.log(Level.WARNING, "Error seeking to " + framePositions[i] + "ms in {0}", ImageUtils.getContentPathSafe(file)); //NON-NLS
+                    // If we can't set the time, continue to the next frame position and try again.
+                    continue;
+                }
+                // Read the frame into the image/matrix.
+                if (!videoFile.read(imageMatrix)) {
+                    LOGGER.log(Level.WARNING, "Error reading frame at " + framePositions[i] + "ms from {0}", ImageUtils.getContentPathSafe(file)); //NON-NLS
+                    // If the image is bad for some reason, continue to the next frame position and try again.
+                    continue;
+                }
+                
+                break;
+            }
 
-            for (int x = 0; x < THUMB_COLUMNS; x++) {
-                for (int y = 0; y < THUMB_ROWS; y++) {
-                    if (Thread.interrupted()) {
-                        return null;
-                    }
-                    if (!videoFile.set(CV_CAP_PROP_POS_MSEC, timestamp + x * framkeskip + y * framkeskip * THUMB_COLUMNS)) {
-                        LOGGER.log(Level.WARNING, "Error seeking to " + timestamp + "ms in {0}", ImageUtils.getContentPathSafe(file)); //NON-NLS
-                        break; // if we can't set the time, return black for that frame
-                    }
-                    //read the frame into the image/matrix
-                    if (!videoFile.read(imageMatrix)) {
-                        LOGGER.log(Level.WARNING, "Error reading frames at " + timestamp + "ms from {0}", ImageUtils.getContentPathSafe(file)); //NON-NLS
-                        break; //if the image for some reason is bad, return black for that frame
-                    }
-
-                    if (bufferedImage == null) {
-                        bufferedImage = new BufferedImage(imageMatrix.cols() * THUMB_COLUMNS, imageMatrix.rows() * THUMB_ROWS, BufferedImage.TYPE_3BYTE_BGR);
-                    }
-
-                    byte[] data = new byte[imageMatrix.rows() * imageMatrix.cols() * (int) (imageMatrix.elemSize())];
-                    imageMatrix.get(0, 0, data); //copy the image to data
-
-                    //todo: this looks like we are swapping the first and third channels.  so we can use  BufferedImage.TYPE_3BYTE_BGR
-                    if (imageMatrix.channels() == 3) {
-                        for (int k = 0; k < data.length; k += 3) {
-                            byte temp = data[k];
-                            data[k] = data[k + 2];
-                            data[k + 2] = temp;
-                        }
-                    }
-
-                    bufferedImage.getRaster().setDataElements(imageMatrix.cols() * x, imageMatrix.rows() * y, imageMatrix.cols(), imageMatrix.rows(), data);
+            // Convert the matrix that contains the frame to a buffered image.
```

Comment:
You might consider checking the height and width of the imageMatrix here. Otherwise we end up with log messages and stack traces related to bad thumbnail dimensions.

Reply:
I've added a call to 'empty()'. If true, we simply return null.